src/jit/gtlist.h

```diff
@@ -59,6 +59,8 @@ GTNODE(XCHG             , GenTreeOp          ,0,GTK_BINOP)
 GTNODE(CMPXCHG          , GenTreeCmpXchg     ,0,GTK_SPECIAL)
 GTNODE(MEMORYBARRIER    , GenTree            ,0,(GTK_LEAF|GTK_NOVALUE))
 
+GTNODE(KEEPALIVE        , GenTree            ,0,(GTK_UNOP|GTK_NOVALUE)) // keep operand alive, generate no code, produce no result
```

Comment:
Is it worth a new node type for this?

If we did some minimal refactoring we could just track this as a `GenTreeJitIntrinsic` with the right `gtIntrinsicId` (which seems more generally useful).
* The only "blocker" on this today is that `GenTreeSimd` inherits from `GenTreeJitIntrinsic` and uses `SIMDIntrinsicID` rather than `NamedIntrinsic`. We could minimally use a typedef and just move the `#define SIMD_INTRINSIC(); #include "simdintrinsiclist.h"` into `namedintrinsiclist.h` to get this cheaply (and then do a full cleanup afterwards).

Reply:
I would say that this has unique characteristics to deserve a special node, and it follows the pattern used for other named similar named intrinsics.

If this cleanup is a good idea, I believe it should be done for all similar named intrinsics: `LOCKADD`, `XADD`, `XCHG`, `CMPXCHG`, `BSWAP`, `BSWAP16`, ...

Reply:
> If we did some minimal refactoring we could just track this as a GenTreeJitIntrinsic with the right gtIntrinsicId (which seems more generally useful).

I'm going to delete `GenTreeJitIntrinsic` as part of the `GT_LIST` removal. It currently serves no real purpose and it would be preferable to continue to do so until then.

Reply:
We should probably discuss in a new issue, but there might be some benefit to keeping it around in some capacity.

There is still quite a bit of logic that is currently duplicated between `GenTreeSIMD` and `GenTreeHWIntrinsic`, with `GenTreeJitIntrinsic` serving as a common base type. `GenTreeJitIntrinsic` will allow more of that logic to be shared.

Reply:
> There is still quite a bit of logic that is currently duplicated between GenTreeSIMD and GenTreeHWIntrinsic, with GenTreeJitIntrinsic serving as a common base type. GenTreeJitIntrinsic will allow more of that logic to be shared.

The only sane way to deal with the `GenTreeSIMD` and `GenTreeHWIntrinsic` duplication is to get rid of `GenTreeSIMD` and use `GenTreeHWIntrinsic` instead. Of course, that will also result in `GenTreeJitIntrinsic` being even more useless than it is today.